pasqal_cloud/__init__.py

```
@@ -59,14 +59,13 @@ class SDK:
 
     def __init__(
         self,
+        project_id: str,
         username: Optional[str] = None,
         password: Optional[str] = None,
         token_provider: Optional[TokenProvider] = None,
         endpoints: Optional[Endpoints] = None,
         auth0: Optional[Auth0Conf] = None,
         webhook: Optional[str] = None,
```

mention this in the changelog since a common way before of instantiating the sdk would have been
```python
sdk = SDK("awennersteen", "super-secret", project_id="my-project_id")
```
And this won't work anymore.

---

Indeed, bad python practices do not help haha, I will update the changelog for this potential breaking change.

---

I'm not sure I understand why you decided to move the project_id to the first argument

---

-> non-default parameter follows default parameter

Python things if I correctly understood how it works

---

It's probably not worth the risk of breaking scripts of user to make it non-optional

---

If this is going to cause significant breaks, we could accommodate by still defaulting to `None` and keep on raising an exception in the scenario it's missing.

We have optional input for several fields that are very obviously not optional fields. Can't authenticate without a password.

---

Then I will put it back as optional and add a warning if not given. Thx for the feedback.

---

You should raise an exception not only a warning!

---

I meant, an exception for sure, but also a warning saying that project_id will be the first argument later, like a deprecation warning.

---

I'd avoid the warning, we should avoid preemptive statements because nothings designed or planned.

What if we simply change our mind on something?